presto-main/src/main/java/com/facebook/presto/security/KerberosPrincipalAccessControlRule.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.security;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class KerberosPrincipalAccessControlRule
+{
+    private final boolean exactMatch;
+
+    @JsonCreator
+    public KerberosPrincipalAccessControlRule(
+            @JsonProperty("exactMatch") boolean exactMatch)
+    {
+        this.exactMatch = exactMatch;
+    }
+
+    public boolean match(String user, String kerberosUserName)
+    {
+        return !exactMatch || user.equals(kerberosUserName);
```

Comment:
Kerberos principals are case sensitive, unfortunately. It's unclear how that should interact with Presto in which users aren't case sensitive, but enforcing case sensitive equals seems best here.